### PR TITLE
Document constraint primal better

### LINKED ISSUE #220: document "constraint primal" better
@tkoolen

### PR DESCRIPTION
**Pull request by odow**
Closes #220

### CODE REVIEW

File: src/attributes.jl [hunk 1]
```diff
@@ -488,6 +488,11 @@ struct ConstraintDualStart <: AbstractConstraintAttribute end
 
 The assignment to the constraint primal values in result `N`.
 If `N` is omitted, it is 1 by default.
+
+Given a constraint `function-in-set`, the `ConstraintPrimal` is the value of the
+function evaluated at the primal solution of the variables. For example, given
+the constraint `2x + y <= 1`, and a primal solution of `(x,y) = (1,2)`, the
```
Comment (mlubin) [inline src/attributes.jl]: There are multiple ways to encode this constraint in MOI, could help to be more precise (ScalarAffineFunction).

File: src/attributes.jl [hunk 2]
```diff
@@ -488,6 +488,12 @@ struct ConstraintDualStart <: AbstractConstraintAttribute end
 
 The assignment to the constraint primal values in result `N`.
 If `N` is omitted, it is 1 by default.
+
+Given a constraint `function-in-set`, the `ConstraintPrimal` is the value of the
+function evaluated at the primal solution of the variables. For example, given
+the constraint `ScalarAffineFunction([x,y], [1, 2], 3)`-in-`LessThan(0)` and
```
Comment (blegat) [inline src/attributes.jl]: Maybe an example with `LessThan(4)` would be better to show that this value is not used in the constraint primal.

File: src/attributes.jl [hunk 3]
```diff
@@ -488,6 +488,12 @@ struct ConstraintDualStart <: AbstractConstraintAttribute end
 
 The assignment to the constraint primal values in result `N`.
 If `N` is omitted, it is 1 by default.
+
+Given a constraint `function-in-set`, the `ConstraintPrimal` is the value of the
+function evaluated at the primal solution of the variables. For example, given
+the constraint `ScalarAffineFunction([x,y], [1, 2], 3)`-in-`LessThan(4)` and
```
[the remark this answers is not in the document]
Comment (odow) [reply src/attributes.jl]: Oh, in LQOI we move them over https://github.com/JuliaOpt/LinQuadOptInterface.jl/blob/bbe798a1377b85453f390e0fa2f87fa5425e345a/src/constraints/scalaraffine.jl#L76

[the remark this answers is not in the document]
Comment (mlubin) [reply src/attributes.jl]: http://www.juliaopt.org/MathOptInterface.jl/stable/apimanual.html#JuMP-mapping-1
That said, from the perspective of JuMP, solvers can safely choose to not support the following constraints:

- ScalarAffineFunction in GreaterThan, LessThan, or EqualTo with a nonzero constant in the function. Constants in the affine function should instead be moved into the parameters of the corresponding sets.

[the remark this answers is not in the document]
Comment (odow) [reply src/attributes.jl]: I changed it to the Interval set instead.

**Pull request merged**